### PR TITLE
Assemblyscript support

### PR DESCRIPTION
**Pull request by ilackarms**
updates `init` and `build` to support `assemblyscript`. drops istio 1.4.2 as a valid target (as we couldn't get AS working on it).

updates the e2e test to build an assemblyscript test image (cuz it builds frickin fast)
BOT NOTES: 
resolves https://github.com/solo-io/wasme/issues/71

### CODE REVIEW

File: docs/content/reference/cli/wasme.md [hunk 1]
```diff
@@ -13,24 +13,18 @@ weight: 5
...
 ### SEE ALSO
 
-* [wasme build](../wasme_build)	 - Build a wasm image from the filter source directory using Bazel-in-Docker
-* [wasme catalog](../wasme_catalog)	 - interact with catalog
+* [wasme build](../wasme_build)	 - Build a wasm image from the filter source directory.
 * [wasme deploy](../wasme_deploy)	 - Deploy an Envoy WASM Filter to the data plane (Envoy proxies).
 * [wasme init](../wasme_init)	 - Initialize a project directory for a new Envoy WASM Filter.
 * [wasme list](../wasme_list)	 - List Envoy WASM Filters stored locally or published to webassemblyhub.io.
-* [wasme login](../wasme_login)	 - login so you can push images to webassemblyhub.io and submit them to the curated catalog
+* [wasme login](../wasme_login)	 - Log in in so you can push images to the remote server.
```
Comment (yuval-k) [inline docs/content/reference/cli/wasme.md]: log in in

File: docs/content/reference/cli/wasme_build_assemblyscript.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+Build a wasm image from an AssemblyScript filter using NPM-in-Docker
+
+```
+wasme build assemblyscript SOURCE_DIRECTORY [-b <bazel target>] -t <name:tag> [flags]
+```
+
+### Options
+
+```
+  -e, --email string      Email for logging in to NPM before running npm install. Optional
```
Comment (yuval-k) [inline docs/content/reference/cli/wasme_build_assemblyscript.md]: why are these here?

File: docs/content/reference/cli/wasme_build_cpp.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+```
+  -c, --config string      The path to the filter configuration file for the image. If not specified, defaults to <SOURCE_DIRECTOR>/runtime-config.json. This file must be present in order to build the image.
+  -d, --debug              debug mode
+  -i, --image string       Name of the docker image containing the Bazel run instructions. Modify to run a custom builder image (default "quay.io/solo-io/ee-builder:dev")
+      --store string       Set the path to the local storage directory for wasm images. Defaults to $HOME/.wasme/store
+  -t, --tag string         The image ref with which to tag this image. Specified in the format <name:tag>. Required
+      --tmp-dir string     Directory for storing temporary files during build. Defaults to /tmp on OSx and Linux. If unset, temporary files will be removed after build
+  -v, --verbose            verbose output
+      --wasm-file string   If specified, wasme will use the provided path to a compiled filter wasm to produce the image. The bazel build will be skipped and the wasm-file will be used instead.
```
Comment (yuval-k) [inline docs/content/reference/cli/wasme_build_cpp.md]: i think we can make this a non persistent flag, so it dosent show up in here

File: pkg/cache/deploy.go [hunk 4]
```diff
@@ -169,9 +169,10 @@ func (d *deployer) createOrUpdateDaemonSet() error {
 						},
 					},
 					Containers: []v1.Container{{
-						Name:  d.name,
-						Image: d.image,
-						Args:  d.args,
+						Name:            d.name,
+						Image:           d.image,
+						ImagePullPolicy: v1.PullAlways,
```
Comment (yuval-k) [inline pkg/cache/deploy.go]: pull always should only be done in ci right? should we gate this in an env var?

File: pkg/cmd/list/list.go [hunk 5]
```diff
@@ -153,7 +153,10 @@ func getLocalImages(storageDir string) ([]image, error) {
 	}
 
 	sort.Slice(images, func(i, j int) bool {
-		return images[i].name < images[j].name
+		if images[i].name < images[j].name {
+			return true
+		}
+		return images[i].updated.Before(images[j].updated)
```
Comment (yuval-k) [inline pkg/cmd/list/list.go]: your probably want
```

		if images[i].name < images[j].name {
			return true
		}
		if images[i].name > images[j].name {
			return false
		}
		return images[i].updated.Before(images[j].updated) 
```

File: docs/content/reference/cli/wasme_build_assemblyscript.md (hunk 2 above)
Comment (ilackarms) [reply docs/content/reference/cli/wasme_build_assemblyscript.md]: for running `npm install` inside the container in case the user needs to.

i think a token would be better here, i can address in a follow-up

File: docs/content/reference/cli/wasme_build_cpp.md (hunk 3 above)
Comment (ilackarms) [reply docs/content/reference/cli/wasme_build_cpp.md]: it's actually not a relevant flag anymore, i forgot to remove from the opts. ty

File: pkg/cmd/list/list.go (hunk 5 above)
Comment (ilackarms) [reply pkg/cmd/list/list.go]: ty! good catch
